src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -181,17 +181,36 @@ void ICollection.CopyTo(Array array, int index)
         }
 
         // Adds item to the tail of the queue.
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void Enqueue(T item)
         {
-            if (_size == _array.Length)
+            int size = _size;
+            T[] array = _array;
+
+            if ((uint)size < (uint)array.Length)
             {
-                int newcapacity = (int)((long)_array.Length * (long)GrowFactor / 100);
-                if (newcapacity < _array.Length + MinimumGrow)
-                {
-                    newcapacity = _array.Length + MinimumGrow;
-                }
-                SetCapacity(newcapacity);
+                _array[_tail] = item;
+                MoveNext(ref _tail);
+                _size++;
```

Comment:
```suggestion
                _size = size + 1;
```
`size` is already in a register, so it just needs to be incremented and written to memory.
Otherwise it is a load, inc, store, so one memory access more, that can be omitted.